library/spdm_transport_storage_lib/libspdm_storage.c

```diff
@@ -0,0 +1,819 @@
+/**
+ *  Copyright Notice:
+ *  Copyright 2024 DMTF. All rights reserved.
+ *  License: BSD 3-Clause License. For full text see link: https://github.com/DMTF/libspdm/blob/main/LICENSE.md
+ **/
+
+/**
+ *  Author: Wilfred Mallawa <wilfred.mallawa@wdc.com>
+ *          Alistair Francis <alistair.francis@wdc.com>
+ **/
+
+#include "library/spdm_transport_storage_lib.h"
+#include "industry_standard/storage.h"
+#include "internal/libspdm_common_lib.h"
+#include "hal/library/debuglib.h"
+#include "hal/library/memlib.h"
+
+# define BSWAP32(x)  ((((x) & 0x000000ff) << 24) |      \
```

Comment:
https://github.com/DMTF/libspdm/blob/b327ead227f14998789bca7d045af17b99dd6d50/include/internal/libspdm_common_lib.h#L1705 already exists in a header. I would rename these to `libspdm_byte_swap_*` and place them there. `libspdm_le_to_be_64` can be renamed to `libspdm_byte_swap_64` in a subsequent pull request to make things consistent.